keps/sig-node/3570-cpumanager/README.md

```diff
@@ -323,6 +407,245 @@ func (p *staticPolicy) RemoveContainer(s State, containerID string) error {
+
+Yes, using the kubelet config.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The impact is node-local only.
+If the state of a node is steady, no changes.
+If a guaranteed pod is admitted, running non-guaranteed pods will have their CPU cgroup changed while running.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, covered by e2e tests
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+A rollout can fail if a bug in the cpumanager prevents _new_ pods to start, or existing pods to be restarted.
+Already running workload will not be affected if the node state is steady
+
+###### What specific metrics should inform a rollback?
+
+"cpu_manager_pinning_errors_total". It must be noted that even in fully healthy system there are known benign condition
+that can cause CPU allocation failures. Few selected examples are:
+
+- requesting odd numbered cores (not a full physical core) when the cpumanager is configured with the `full-pcpus-only` option
+- requesting NUMA-aligned cores, with Topology Manager enabled.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+No to both.
+Changes in behavior only affects pods meeting the conditions (guaranteed QoS, integral CPU request) scheduler after the upgrade.
+Running pods will be unaffected by any change. This offers some degree of safety in both upgrade->rollback
+and upgrade->downgrade->upgrade scenarios.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+Monitor the pod admission counter
+Monitor the pods not going running after successful schedule
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The operator need to inspect the node and verify the cpu pinning assignment either checking the cgroups on the node
+or accessing the podresources API of the kubelet.
+
+###### How can someone using this feature know that it is working for their instance?
+
+
+- [X] Other (treat as last resort)
+  - Details: the containers need to check the cpu set they are allowed to run; in addition, node agents (e.g. node_exporter)
+    can report the CPU assignment
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+- N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [ ] Other (treat as last resort)
+  - Details:
+     a operator should check that pods go running correctly and the cpu pinning is performed. The latter can
+     be checked by inspecting the cgroups at node level.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+"cpu_manager_pinning_requests_total" and "cpu_manager_pinning_errors_total"
```

Comment:
This won't be missing when GA, right? Can you move this into the "What are the reasonable SLOs (Service Level Objectives) for the enhancement"?  We will trust you to add them

Reply:
I'm 100% committed in having them in: [issue](https://github.com/kubernetes/kubernetes/issues/112854) and [PR](https://github.com/kubernetes/kubernetes/pull/112855) are up to already. From my PoV the only thing missing in the PR is some form of testing to ensure the metrics are populated.